iceoryx_utils/test/moduletests/test_posix_signal_handler.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_utils/posix_wrapper/signal_handler.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::posix;
+
+int signalOfCallback1 = 0;
+int signalOfCallback2 = 0;
```

Comment:
Shouldn't this be atomics? I'm not quite sure, but since it is set from a different context it might be necessary.

Reply:
I am also not quite sure. I think the kernel causes the signalHandler to be executed in the thread which receives the signal, so an atomic would not be necessary here. A quick search confirms this, but it is not a reliable source https://stackoverflow.com/questions/6949025/how-are-asynchronous-signal-handlers-executed-on-linux.

Would not hurt to make those atomic I guess, then we would be sure the updates will be visible here as well (after some cache syncs at least....).